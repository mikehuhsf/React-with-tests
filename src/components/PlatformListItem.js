import React, { useState } from "react";

const PlatformListItem = ({ platform }) => {
  const [GamesList, setGamesList] = useState(null);

  const GQL_API = `http://localhost:3030/`;
  const GQL_QUERY = `
    query($id: ID!){
      platform(id: $id){
        name
        release_date
        specifications
      }
    }`;

  const handleLoadGames = () => {
    const variables = { id: platform.id };
    fetch(GQL_API, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        query: GQL_QUERY,
        variables,
      }),
    })
      .then((response) => response.json())
      .then((result) => setGamesList(result.data.Platform.games));
  };

  return (
    <div>
      <a href="#" onClick={handleLoadGames}>
        {platform.name}
      </a>
      {GamesList &&
        GamesList.map((Game) => {
          return <div key={Game.id}>{Game.name}</div>;
        })}
    </div>
  );
};

//export default PlatformListItem;
