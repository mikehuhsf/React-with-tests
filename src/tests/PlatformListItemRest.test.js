import React from 'react';
import renderer from 'react-test-renderer';
import PlatformListItem from "../components/PlatformListItemRest";
import { specifications } from '../tools/mockSpecifications';
import { shallow } from '../enzyme';

it("renders a drop down when clicked", () => {
    const wrapper = renderer.create(<PlatformListItem name='iPhone13' release_date='2021-09-24' specifications={specifications}/>);
    expect(wrapper).toMatchSnapshot();
})

it('should find 1 href named name', () => {
    const wrapper = shallow(<PlatformListItem />);
    expect(wrapper.find('.button-increment').text()).toEqual('Increment');

  });