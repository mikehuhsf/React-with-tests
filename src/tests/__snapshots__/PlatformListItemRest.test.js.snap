// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a drop down whenn clicked 1`] = `
<div>
  <a
    href="#"
    onClick={[Function]}
  >
    iPhone13
  </a>
</div>
`;
